ynr/apps/candidates/forms.py

```diff
@@ -23,14 +23,25 @@ def clean_assigned_to_dc(self):
 
 
 class ToggleLockForm(forms.ModelForm):
+
+    hashed_memberships = forms.CharField(
+        widget=forms.HiddenInput, required=False
+    )
+    lock = forms.CharField(widget=forms.HiddenInput(), required=False)
+    candidates_locked = forms.HiddenInput()
+
     class Meta:
         model = Ballot
         fields = ("candidates_locked",)
 
-    candidates_locked = forms.HiddenInput()
-
     def clean(self):
-        if self.data.get("lock") == "lock":
+        if self.cleaned_data.get("lock") == "lock":
+            if (
+                self.cleaned_data["hashed_memberships"]
+                != self.instance.hashed_memberships
+            ):
+                raise ValidationError("Ballot has changed, please recheck")
```

Comment:
Can these errors contain something along the line of "by someone else since loading" - I don't know if that's the ideal wording, but it would be nice to explain _how_ the ballot changed (and that they've done nothing wrong)

Reply:
Good idea - made them a bit clearer that another user has changed the ballot:
<img width="1552" alt="Screenshot 2021-03-17 at 14 29 23" src="https://user-images.githubusercontent.com/15347726/111484005-41136400-872d-11eb-94d9-27f6e34043c7.png">
<img width="1552" alt="Screenshot 2021-03-17 at 14 29 11" src="https://user-images.githubusercontent.com/15347726/111484012-440e5480-872d-11eb-90fa-69204e6d61a6.png">